subprojects/docs/src/docs/userguide/declaringDependencies.adoc

```diff
@@ -179,6 +183,32 @@ Let's say you wanted to declare a dependency on the link:https://tomcat.apache.o
 
 A project's configurations are managed by a `configurations` object. Configurations have a name and can extend each other. To learn more about this API have a look at api:org.gradle.api.artifacts.ConfigurationContainer[].
 
+[[sub:inheriting_dependencies_from_other_configurations]]
+==== Inheriting dependencies from other configurations
+
+A configuration can extend other configurations to form an inheritance hierarchy similar to class inheritance in object-oriented programming. Configurations with a super configuration have access to the whole set of dependencies assigned to the hierarchy.
+
+This concept is used by Gradle core plugins like the <<sec:java_plugin_and_dependency_management,Java plugin>>. For example the `testCompile` configuration extends the `compile` configuration. Compiling tests requires the dependencies of the source code under test to be available on the classpath on top of the dependencies needed to compile the test code. You could easy imagine a Java project that uses the Guava dependency in the production source code and JUnit to write and execute test code.
+
++++++
+<figure xmlns:xi="http://www.w3.org/2001/XInclude">
+    <title>Configuration inheritance provided by the Java plugin</title>
+    <imageobject>
+        <imagedata fileref="img/dependency-management-configuration-inheritance.png" width="140mm" />
```

Comment:
❌ Should use implementation/testImplementation, not the deprecated compile/testCompile

Reply:
done. We'll need to do something about the documentation on that front. It's absolutely unclear to a user that these configurations are deprecated or "somewhat, lightly deprecated". The documentation should say "Don't use these because of x. Use y instead.". Are there still _any_ uses cases where you'd want to use `compile` or `testCompile`?

Reply:
I think we should continue the discussion here and address this as part of the "Update Java (Library) plugin chapter" story: https://github.com/gradle/dotorg-docs/issues/42